xwiki-platform-core/xwiki-platform-image/xwiki-platform-image-style/xwiki-platform-image-style-api/src/main/java/org/xwiki/image/style/internal/rendering/CaptionedImageParseListenerProvider.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.image.style.internal.rendering;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.rendering.listener.ListenerProvider;
+import org.xwiki.rendering.listener.QueueListener;
+import org.xwiki.rendering.listener.chaining.ChainingListener;
+import org.xwiki.rendering.listener.chaining.EventType;
+import org.xwiki.rendering.listener.chaining.ListenerChain;
+import org.xwiki.rendering.listener.chaining.LookaheadChainingListener;
+import org.xwiki.rendering.listener.reference.ResourceReference;
+import org.xwiki.rendering.syntax.Syntax;
+
+import static org.xwiki.image.style.internal.rendering.CaptionedImageRenderListenerProvider.DATA_XWIKI_IMAGE_STYLE;
+import static org.xwiki.image.style.internal.rendering.CaptionedImageRenderListenerProvider.DATA_XWIKI_IMAGE_STYLE_ALIGNMENT;
+import static org.xwiki.image.style.internal.rendering.CaptionedImageRenderListenerProvider.DATA_XWIKI_IMAGE_STYLE_BORDER;
+import static org.xwiki.image.style.internal.rendering.CaptionedImageRenderListenerProvider.DATA_XWIKI_IMAGE_STYLE_TEXT_WRAP;
+import static org.xwiki.image.style.internal.rendering.CaptionedImageRenderListenerProvider.STYLE_PROPERTY;
+import static org.xwiki.image.style.internal.rendering.CaptionedImageRenderListenerProvider.WIDTH_PROPERTY;
+import static org.xwiki.rendering.syntax.Syntax.XWIKI_2_0;
+import static org.xwiki.rendering.syntax.Syntax.XWIKI_2_1;
+
+/**
+ * Provides a parser listener for image with captions, taking into account the image styles.
+ *
+ * @version $Id$
+ * @since 15.3RC1
+ * @since 14.10.8
+ */
+@Component
+@Singleton
+@Named("captionedImageParse")
+public class CaptionedImageParseListenerProvider implements ListenerProvider
+{
+    private static final List<Syntax> ACCEPTED_SYNTAX = List.of(XWIKI_2_0, XWIKI_2_1);
```

Comment:
A Set might be better since the point is to use `#contains` (but not sure for such a small collection honestly).

Reply:
+1 because it's conceptually more correct. Though I agree it won't make much difference in practice.